src/App.tsx

```diff
@@ -1,14 +1,66 @@
-import React from 'react';
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import { DropDownItem } from './components/DropDownItem';
+
+export const useDebounce = (value: string, delay: number) => {
+  const [debouncedValue, setDebouncedValue] = useState(value);
+
+  useEffect(
+    () => {
+      const handler = setTimeout(() => {
+        setDebouncedValue(value);
+      }, delay);
+
+      return () => {
+        clearTimeout(handler);
+      };
+    },
+    [value, delay],
+  );
+
+  return debouncedValue;
+};
+
+const searchedPeople = (peopleData: Person[], searchQuery: string) => {
+  const lowerQuery = searchQuery.trim().toLocaleLowerCase();
+
+  return peopleData.filter(people => people.name
+    .toLowerCase()
+    .includes(lowerQuery));
+};
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [query, setQuery] = useState('');
+
+  const debouncedQuery = useDebounce(query, 1000);
+
+  const onSelect = useCallback(
+    (person: Person) => {
+      setSelectedPerson(person);
+    },
+    [debouncedQuery],
+  );
+
+  const onQuery = useCallback(() => setQuery(''), [debouncedQuery]);
```

Comment:
Why do you need debouncedQuery here in the dependency array? Seems that this function doesn't depend on debouncedQuery. And actually, useCallback is redundant here, so you can remove it at all. Sometimes it's more expensive to use memorization then not